app/org/sagebionetworks/bridge/dynamodb/DynamoUploadSchemaDao.java

```diff
@@ -82,6 +86,101 @@ public void setStudyIdIndex(DynamoIndexHelper studyIdIndex) {
 
     /** {@inheritDoc} */
     @Override
+    public @Nonnull UploadSchema createUploadSchemaFromSurvey(@Nonnull StudyIdentifier studyIdentifier,
+            @Nonnull Survey survey) {
+        // create upload field definitions from survey questions
+        List<UploadFieldDefinition> fieldDefList = new ArrayList<>();
+        for (SurveyQuestion oneQuestion : survey.getUnmodifiableQuestionList()) {
+            String name = oneQuestion.getIdentifier();
+            UploadFieldType type = getFieldTypeFromConstraints(oneQuestion.getConstraints());
+            UploadFieldDefinition oneFieldDef = new DynamoUploadFieldDefinition.Builder().withName(name)
+                    .withType(type).build();
+            fieldDefList.add(oneFieldDef);
+        }
+
+        // Get the current rev.
+        String studyId = studyIdentifier.getIdentifier();
+        String schemaId = survey.getIdentifier();
```

Comment:
Discussed this with Alx over IM. The idea is that the schema ID should be something that researchers can set, and it should match the survey, so survey ID (which is different from survey GUID) is what we want here.